app/controllers/items_controller.rb

```diff
@@ -1,19 +1,39 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_wishlist_and_item, only: %i(edit update)
+  before_action :set_wishlist, only: %i(new create edit update)
+  before_action :set_item, only: %i(edit update)
+
+  def new
+    @item = @wishlist.items.new
+    authorize! :new, @item
+  rescue CanCan::AccessDenied
+    redirect_to @wishlist, alert: t("generic_error")
```

Comment:
Create someone else's wishlist? How can it be done? Try to edit someone else's wishlist and be redirected to this specific wishlist? Which one? Someone else's??